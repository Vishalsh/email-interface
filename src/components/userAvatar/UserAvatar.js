import React from 'react';
import PropTypes from 'prop-types';

import classes from './UserAvatar.module.scss';

const UserAvatar = (props) => {
  const {
    name,
    role,
    avatar
  } = props;

  return (
    <div className={classes.avatar}>
      <img src={avatar} alt={name} className={classes.avatarImage}/>
      <p className={classes.name}>{name}</p>
      <p className={classes.role}>
        <span>{role}</span>
        <i className="icon icon-arrow-down" />
      </p>
    </div>
  );
};

UserAvatar.propTypes = {
  name: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
};

export default UserAvatar;
