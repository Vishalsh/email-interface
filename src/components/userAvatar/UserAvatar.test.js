import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import UserAvatar from './UserAvatar';

describe('UserAvatar', () => {
  const props = {
    name: 'MS Dhoni',
    role: 'Wicket Keeper',
    avatar: 'http://s.ndtvimg.com/images/entities/120/ms-dhoni-700.png'
  };
  const component = shallow(<UserAvatar {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});