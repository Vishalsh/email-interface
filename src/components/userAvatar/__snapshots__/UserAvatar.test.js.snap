// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should render the component 1`] = `
<div
  className="avatar"
>
  <img
    alt="MS Dhoni"
    className="avatarImage"
    src="http://s.ndtvimg.com/images/entities/120/ms-dhoni-700.png"
  />
  <p
    className="name"
  >
    MS Dhoni
  </p>
  <p
    className="role"
  >
    <span>
      Wicket Keeper
    </span>
    <i
      className="icon icon-arrow-down"
    />
  </p>
</div>
`;
